src/arm/cache.c

```diff
@@ -1239,6 +1241,66 @@ void cpuinfo_arm_decode_cache(
 			};
 			break;
 		}
+		case cpuinfo_uarch_neoverse_n1:
+		case cpuinfo_uarch_neoverse_v1:
+		case cpuinfo_uarch_neoverse_n2:
+                {
+                        /*
+                         * ARM Neoverse-n1 Core Technical Reference Manual
+                         * A6.1. About the L1 memory system
+			 *   The L1 memory system consists of separate instruction and data caches. Both have a fixed size of 64KB.
+                         *
+                         * A6.1.1 L1 instruction-side memory system
+                         *   The L1 instruction memory system has the following key features:
+                         *    - Virtually Indexed, Physically Tagged (VIPT), which behaves as a Physically Indexed,
+                         *      Physically Tagged (PIPT) 4-way set-associative L1 data cache.
+                         *    - Fixed cache line length of 64 bytes.
+                         *
+                         * A6.1.2 L1 data-side memory system
+                         *   The L1 data memory system has the following features:
+                         *    - Virtually Indexed, Physically Tagged (VIPT), which behaves as a Physically Indexed,
+                         *      Physically Tagged (PIPT) 4-way set-associative L1 data cache.
+                         *    - Fixed cache line length of 64 bytes.
+                         *    - Pseudo-LRU cache replacement policy.
+                         *
+                         * A7.1 About the L2 memory system
+                         *   The L2 memory subsystem consist of:
+			 *    - An 8-way set associative L2 cache with a configurable size of 256KB, 512KB, or 1024KB. Cache lines
+			 *      have a fixed length of 64 bytes.
+                         *    - Strictly inclusive with L1 data cache.
+			 *    - When configured with instruction cache hardware coherency, strictly inclusive with L1 instruction cache.
+			 *    - When configured without instruction cache hardware coherency, weakly inclusive with L1 instruction cache.
+                         */
+
+			const uint32_t min_l2_size_KB= 256;
+			const uint32_t min_l3_size_KB = 0;
+
+			uint32_t proc_l2_size_KB = cpuinfo_linux_get_processor_cache_size(system_processor_id, 2);
+			uint32_t proc_l3_size_KB = cpuinfo_linux_get_processor_cache_size(system_processor_id, 3);
```

Comment:
This code would break compilation for say iOS or M1 platforms, wouldn't it?

Reply:
Sure, I will split it into 2 PRs.
Regarding the dynamic cache querying feature, having a general framework support is good as it is going to benefit all SoCs, not just the neoverse family.